sample-3.md

```diff
@@ -0,0 +1,548 @@
+# Step-by-step: Deploy "Hello World" Eventing soluction
+
+This step-by-step guide will guide you through all the steps involved to 
+run “Hello World” eventing solution, where you will send an event to the 
+Knative eventing broker inside your K8s namespace and receive it in a pod 
+using Knative eventing. ETA - 20-30 mins. If you want to skip all the steps 
+and want to quickly deploy the solution then please refer to 
+[Quick Start: Deploy "Hello World" eventing solution](todo).
+
+## Before you begin
+
+Before you begin:
+- [Install Knative Eventing](todo) and verify the health of the Knative Eventing
+  control plane.
+  
+## Description
+
+This sample is identical to [Quick Start: Deploy "Hello World" eventing solution](todo),
+but rather than using a single YAML, we will apply each piece one step at a time, 
+explaining what it does as we do so.
+
+We will create the following topology:
+
+Image from 2.
+
+- [`Pod` named `curl`](#event-producer), acting as the event producer.
+- `Broker` named `default`.
+- Two [`Trigger`s](#triggers), each with a mutually exclusivefilter pointing at distinct 
+  subscribers.
+- Two Kubernetes [`Service`s](#event-consumers), each pointing at a unique:
+    - [`Deployment`](#event-consumers), acting as the event consumer.
+
+## Helpers
+
+First we will create useful shell variables to allow us to not have to type 
+things out every time.
+
+```shell
+K_NAMESPACE=kn-eventing-step-by-step-sample
```

Comment:
It might be worth simplifying things by setting this as the default namespace in the active kubectl config context.

Reply:
My personal choice is to not use `kubectl` contexts. I've found that I often confuse myself ('What do you mean that service doesn't exist?!? I just made it!').

But we should let the docs folks determine what is easiest for people to use and do that consistently across all our docs.